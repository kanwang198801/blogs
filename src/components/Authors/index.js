import React, { memo } from 'react';
import Author from '../Author';
import styles from './styles.module.css';

function Authors({ authors }) {
  return (
    <div className={styles.authorsContainer}>
      <h5>Authors </h5>
      <div className={styles.authors}>
        {authors.length > 0 &&
          authors.map((author, index) => (
            <Author author={author} key={index} />
          ))}
      </div>
    </div>
  );
}
export default memo(Authors);
