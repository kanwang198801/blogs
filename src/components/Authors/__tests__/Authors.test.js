import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Authors from '../index';

const authors = [
  {
    name: 'Centr Team',
    imageList: {
      square1x: {
        url:
          'https://cdn.centr.com/content/5000/4738/images/square1x-cen-author-pic-the-expertsv2.jpg',
      },
      square2x: {
        url:
          'https://cdn.centr.com/content/5000/4738/images/square2x-cen-author-pic-the-expertsv2.jpg',
      },
      square3x: {
        url:
          'https://cdn.centr.com/content/5000/4738/images/square3x-cen-author-pic-the-expertsv2.jpg',
      },
    },
  },
  {
    name: 'Tech Team',
    imageList: {
      square1x: {
        url:
          'https://cdn.centr.com/content/5000/4738/images/square1x-cen-author-pic-the-expertsv2.jpg',
      },
      square2x: {
        url:
          'https://cdn.centr.com/content/5000/4738/images/square2x-cen-author-pic-the-expertsv2.jpg',
      },
      square3x: {
        url:
          'https://cdn.centr.com/content/5000/4738/images/square3x-cen-author-pic-the-expertsv2.jpg',
      },
    },
  },
];

afterEach(cleanup);

it('should match the snapshot', () => {
  const { container } = render(<Authors authors={authors} />);
  expect(container.firstChild).toMatchSnapshot();
});
