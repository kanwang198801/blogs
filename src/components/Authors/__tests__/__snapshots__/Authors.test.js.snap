// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  class="authorsContainer"
>
  <h5>
    Authors 
  </h5>
  <div
    class="authors"
  >
    <div
      class="ant-card ant-card-bordered"
      style="width: auto; border: 0px;"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="ant-card-meta"
        >
          <div
            class="ant-card-meta-avatar"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
            >
              <img
                src="https://cdn.centr.com/content/5000/4738/images/square1x-cen-author-pic-the-expertsv2.jpg"
              />
            </span>
          </div>
          <div
            class="ant-card-meta-detail"
          >
            <div
              class="ant-card-meta-title"
            >
              Centr Team
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card ant-card-bordered"
      style="width: auto; border: 0px;"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="ant-card-meta"
        >
          <div
            class="ant-card-meta-avatar"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-image"
            >
              <img
                src="https://cdn.centr.com/content/5000/4738/images/square1x-cen-author-pic-the-expertsv2.jpg"
              />
            </span>
          </div>
          <div
            class="ant-card-meta-detail"
          >
            <div
              class="ant-card-meta-title"
            >
              Tech Team
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
